### PR TITLE
29 update package structure

### LINKED ISSUE #29: update package structure
use the new plugin mechanism.

After doing that, generate new release and write explicitly in the changelog which import must be changed by users in their plugins inheriting from this one

### PR DESCRIPTION
**Pull request by ka-sarthak**
- Updated pyproject, actions, vscode, docs to match with the template https://github.com/FAIRmat-NFDI/nomad-plugin-template
- Removed unnecessary files like nomad yaml and Dockerfile
- Added config for cruft, in case there are changes to be pulled from the template in the future.
- Updated the src code structure: 
  - divided the schemas in general and jupyter (TODO in future: take out general analysis from jupyter)
  - add version file in src init.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -11,31 +11,36 @@ dynamic = ["version"]
 description = "A NOMAD plugin for analysis of FAIR data."
 readme = "README.md"
 authors = [
-    { name = "Sarthak Kapoor", email = "sarthak.kapoor@physik.hu-berlin.de" }
+    { name = "Sarthak Kapoor", email = "sarthakkapoor@hotmail.com" },
```
Comment (andreaa93) [inline pyproject.toml]: maybe this is better 

```suggestion
    { name = "Sarthak Kapoor", email = "sarthak.kapoor@physik.hu-berlin.de" },
```

File: pyproject.toml [hunk 2]
```diff
@@ -11,31 +11,36 @@ dynamic = ["version"]
...
 ]
-license = { text = "Apache-2.0" }
+maintainers = [
+    { name = "Sarthak Kapoor", email = "sarthakkapoor@hotmail.com" },
```
Comment (andreaa93) [inline pyproject.toml]: We will add all the team also in the others

```suggestion
    { name = "Sarthak Kapoor", email = "sarthakkapoor@hotmail.com" },
    { name = "Hampus Näsström", email = 'hampus.naesstroem@physik.hu-berlin.de' },
    { name = "Andrea Albino", email = 'andrea.albino@physik.hu-berlin.de' },
    { name = "Sebastian Brückner", email = 'sebastian.brueckner@ikz-berlin.de' },
```

File: pyproject.toml [hunk 3]
```diff
@@ -46,12 +51,30 @@ dev = [
...
+    # pyupgrade
+    "UP",
+    # isort
+    "I",
+    # pylint
+    "PL",
 ]
-lint.ignore = [
+
+ignore = [
```
Comment (andreaa93) [inline pyproject.toml:79]: we should also harmonize this in the three plugins. Should be ignore all of these? Okay for me

File: src/nomad_analysis/__init__.py [hunk 4]
```diff
@@ -1,20 +1,4 @@
-#
```
Comment (andreaa93) [inline src/nomad_analysis/__init__.py:1]: do we need this init file in the root? I generally do not use it

File: src/nomad_analysis/jupyter/schema.py [hunk 5]
```diff
@@ -65,14 +65,14 @@
...
-        EntryArchive,
         ArchiveSection,
+        EntryArchive,
     )
     from structlog.stdlib import (
         BoundLogger,
     )
 
-m_package = Package(name='analysis_jupyter')
+m_package = Package()
```
Comment (andreaa93) [inline src/nomad_analysis/jupyter/schema.py]: You need to import schemapackage instead of package

```suggestion
m_package = SchemaPackage()
```

Comment (andreaa93) [reply src/nomad_analysis/jupyter/schema.py]: Import it at line 58, I cannot add it from this review

**Review by andreaa93 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (ka-sarthak) [reply pyproject.toml:79]: yea, we can align with the template repo.

File: src/nomad_analysis/__init__.py (hunk 4 above)
Comment (ka-sarthak) [reply src/nomad_analysis/__init__.py:1]: No, this is only needed to have the version of the package also loaded when the package is imported. 
That is, to make this work 
```py
import nomad_analysis
print(nomad_analysis.__version__)
```

File: src/nomad_analysis/jupyter/schema.py (hunk 5 above)
Comment (ka-sarthak) [reply src/nomad_analysis/jupyter/schema.py]: I see, will change it

File: src/nomad_analysis/__init__.py (hunk 4 above)
Comment (andreaa93) [reply src/nomad_analysis/__init__.py:1]: my bad, it should be there

File: pyproject.toml (hunk 3 above)
Comment (andreaa93) [reply pyproject.toml:79]: so the repo only has F403, will you leave only that one?

Comment (ka-sarthak) [reply pyproject.toml:79]: yes

Comment (ka-sarthak) [reply pyproject.toml:79]: If you modify this in another repo, you might have to run a `ruff check .` and perform some formatting that might have been ignored earlier.

**Review by andreaa93 — APPROVED** [review]
very precise! Go ahead

File: pyproject.toml (hunk 3 above)
Comment (andreaa93) [reply pyproject.toml:79]: ah, I see..  makes sense, good to know!

**Pull request merged**